src/giskardpy/plugin_pybullet.py

```diff
@@ -50,61 +51,119 @@ def setup(self, timeout=5.0):
+                    f.write("#add {}\n".format(object_name))
+                    with open("{}/{}.urdf".format(folder_path, object_name), u'w') as f_urdf:
+                        f_urdf.write(world_object.original_urdf)
+
+                    f.write('with open(\'{}/{}.urdf\', \"r\") as f:\n'.format(folder_path,object_name))
+                    f.write("   {}_urdf = f.read()\n".format(object_name))
+                    f.write("{0}_name = \"{0}\"\n".format(object_name))
+                    f.write("{0}_js_topic = \"{0}_js_topic\"\n".format(object_name))
+                    f.write("{}_pose = PoseStamped()\n".format(object_name))
+                    f.write("{}_pose.header.stamp = rospy.Time.now()\n".format(object_name))
+                    f.write("{0}_pose.pose.position = Point({1}, {2}, {3})\n".format(object_name,
+                                                                                     world_object.base_pose.position.x,
+                                                                                     world_object.base_pose.position.y,
+                                                                                     world_object.base_pose.position.z,))
+                    f.write("{0}_pose.pose.orientation = Quaternion({1}, {2}, {3}, {4})\n".format(object_name,
+                                                                                                world_object.base_pose.orientation.x,
+                                                                                                world_object.base_pose.orientation.y,
+                                                                                                world_object.base_pose.orientation.z,
+                                                                                                world_object.base_pose.orientation.w))
+
+                    f.write("{}_pose.header.frame_id = \"map\"\n".format(object_name))
+                    f.write("zero_pose.add_urdf(name={0}_name, urdf={0}_urdf, pose={0}_pose, js_topic={0}_js_topic, set_js_topic=None)\n".format(object_name))
+                    f.write("{}_joint_state = ".format(object_name))
                     write_dict(to_joint_state_position_dict((dict_to_joint_states(world_object.joint_state))), f)
-                    f.write(u'{} base pose\n'.format(object_name))
-                    write_dict(convert_ros_message_to_dictionary(world_object.base_pose),f)
-                    object_links = world_object.get_link_names()
-                    if len(object_links) == 1:
-                        f.write(u'{} shape\n'.format(object_name))
-                        f.write(str(world_object.get_urdf_link(world_object.get_link_names()[0])))
-                        f.write(u'\n')
-            logging.loginfo(u'saved dump to {}'.format(new_path))
+                    f.write("zero_pose.set_object_joint_state({0}_name, {0}_joint_state)\n\n".format(object_name))
```

Comment:
I seems like adding a urdf and setting its joint state directly after, doesn't give the giskards enough time to initialize the subscriber. please add a sleep after the subscriber get created in add_object of pybullet plugin.